cli: fix `download` segfaults when `outputs` directive is wrong

**Current behaviour**

`reana-client-go` segfaults when the outputs directive is wrong.

For example, remove it:

```console
$ cd reana-demo-helloworld
$ vim reana.yaml
$ git diff
diff --git a/reana.yaml b/reana.yaml
index 0fc4418..2d1eb36 100644
--- a/reana.yaml
+++ b/reana.yaml
@@ -18,6 +18,3 @@ workflow:
               --inputfile "${inputfile}"
               --outputfile "${outputfile}"
               --sleeptime ${sleeptime}
-outputs:
-  files:
-   - results/greetings.txt
```

Run such a workflow:

```console
$ reana-dev run-example -c r-d-helloworld -w serial -b kubernetes --submit-only
```

Try to download all outputs:

```console
$ rcg download -w helloworld-serial-kubernetes.3
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xb057b6]

goroutine 1 [running]:
reanahub/reana-client-go/cmd.(*downloadOptions).run(0xc0003ae390, 0x0?, {0xc0002c3580?, 0x0?, 0x0?})
        .../reana-client-go/cmd/download.go:98 +0x76
reanahub/reana-client-go/cmd.newDownloadCmd.func1(0xc0003b0a00?, {0xc0002c3580?, 0x2?, 0x2?})
        .../reana-client-go/cmd/download.go:63 +0x31
github.com/spf13/cobra.(*Command).execute(0xc0003b0a00, {0xc0002c3560, 0x2, 0x2})
        /home/tibor/private/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc0003a6280)
        /home/tibor/private/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3bd
github.com/spf13/cobra.(*Command).Execute(0xc0000061a0?)
        /home/tibor/private/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918 +0x19
main.main()
        .../reana-client-go/main.go:28 +0x25
```

**Expected behaviour**

The `reana-client-go` should not segfault but emit an output message to the user that there is nothing to download and explain why.

--- a/cmd/download.go
+++ b/cmd/download.go
@@ -1,6 +1,6 @@
 /*
 This file is part of REANA.
-Copyright (C) 2022 CERN.
+Copyright (C) 2022, 2023 CERN.
 
 REANA is free software; you can redistribute it and/or modify it
 under the terms of the MIT License; see LICENSE file for more details.
@@ -95,9 +95,10 @@ func (o *downloadOptions) run(cmd *cobra.Command, args []string) error {
 		if err != nil {
 			return err
 		}
-		downloadFiles := spec.Specification.Outputs.Files
-		downloadDirs := spec.Specification.Outputs.Directories
-		downloadPaths = append(downloadFiles, downloadDirs...)
+		if outputs := spec.Specification.Outputs; outputs != nil {
+			downloadPaths = append(downloadPaths, outputs.Files...)
+			downloadPaths = append(downloadPaths, outputs.Directories...)
+		}
 	}
 	log.Debugf("Download paths: %s", strings.Join(downloadPaths, ", "))
 

--- a/cmd/download_test.go
+++ b/cmd/download_test.go
@@ -1,6 +1,6 @@
 /*
 This file is part of REANA.
-Copyright (C) 2022 CERN.
+Copyright (C) 2022, 2023 CERN.
 
 REANA is free software; you can redistribute it and/or modify it
 under the terms of the MIT License; see LICENSE file for more details.
@@ -44,6 +44,15 @@ func TestFileDownload(t *testing.T) {
 			expected: []string{
 				fmt.Sprintf("%s was successfully downloaded.", fileName),
 			},
+		},
+		"download from workflow without outputs": {
+			serverResponses: map[string]ServerResponse{
+				fmt.Sprintf(downloadWorkflowSpecServerPath, "my_workflow"): {
+					statusCode:   http.StatusOK,
+					responseFile: "workflow_specification_without_outputs.json",
+				},
+			},
+			args: []string{"-w", "my_workflow"},
 		},
 		"download file specified as argument": {
 			serverResponses: map[string]ServerResponse{

--- a/testdata/inputs/workflow_specification_without_outputs.json
+++ b/testdata/inputs/workflow_specification_without_outputs.json
@@ -1,0 +1,40 @@
+{
+  "parameters": {},
+  "specification": {
+    "inputs": {
+      "files": [
+        "code/gendata.C",
+        "code/fitdata.C"
+      ],
+      "parameters": {
+        "data": "results/data.root",
+        "events": 20000,
+        "plot": "results/plot.png"
+      }
+    },
+    "version": "0.6.0",
+    "workflow": {
+      "specification": {
+        "steps": [
+          {
+            "commands": [
+              "mkdir -p results && root -b -q 'code/gendata.C(${events},\"${data}\")'"
+            ],
+            "environment": "reanahub/reana-env-root6:6.18.04",
+            "kubernetes_memory_limit": "256Mi",
+            "name": "gendata"
+          },
+          {
+            "commands": [
+              "root -b -q 'code/fitdata.C(\"${data}\",\"${plot}\")'"
+            ],
+            "environment": "reanahub/reana-env-root6:6.18.04",
+            "kubernetes_memory_limit": "256Mi",
+            "name": "fitdata"
+          }
+        ]
+      },
+      "type": "serial"
+    }
+  }
+}

